### PR TITLE
feat(ec2): Client VPN Endpoints Should Have Client Connection Logging Enabled

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

Ensure Client VPN endpoints have client connection logging enabled. The control fails if logging is not enabled for the endpoint. Client connection logs are crucial for tracking user activity and ensuring visibility into VPN usage.

Enabling connection logging helps in monitoring, auditing, and troubleshooting VPN connections, contributing to overall security and compliance.

### Description

I added `ec2_client_vpn_endpoint_connection_logging_enabled` check with respective unit tests. In order to that I modified `ec2_service.py` to add a new model `VpnEndpoint` and added respective test to `ec2_service_test.py`.
### Checklist

- Are there new checks included in this PR? Yes 
    - If so, do we need to update permissions for the provider? No
- [X] Review if the code is being covered by tests.
- [X] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/ec2/ec2_client_vpn_endpoint_connection_logging_enabled/ec2_client_vpn_endpoint_connection_logging_enabled.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        for vpn_arn, vpn_endpoint in ec2_client.vpn_endpoints.items():
+            report = Check_Report_AWS(self.metadata())
+            report.region = vpn_endpoint.region
+            report.resource_id = vpn_endpoint.id
+            report.resource_arn = vpn_arn
+            report.resource_tags = vpn_endpoint.tags
+
+            if vpn_endpoint.connection_logging:
+                report.status = "PASS"
+                report.status_extended = f"Vpn Endpoint {vpn_endpoint.id} in region {vpn_endpoint.region} has client connection logging enabled."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ec2/ec2_client_vpn_endpoint_connection_logging_enabled/ec2_client_vpn_endpoint_connection_logging_enabled.py]:
```suggestion
                report.status_extended = f"Client VPN endpoint {vpn_endpoint.id} in region {vpn_endpoint.region} has client connection logging enabled."
```

File: prowler/providers/aws/services/ec2/ec2_client_vpn_endpoint_connection_logging_enabled/ec2_client_vpn_endpoint_connection_logging_enabled.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            else:
+                report.status = "FAIL"
+                report.status_extended = f"Vpn Endpoint {vpn_endpoint.id} in region {vpn_endpoint.region} does not have client connection logging enabled."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ec2/ec2_client_vpn_endpoint_connection_logging_enabled/ec2_client_vpn_endpoint_connection_logging_enabled.py]:
```suggestion
                report.status_extended = f"Client VPN endpoint {vpn_endpoint.id} in region {vpn_endpoint.region} does not have client connection logging enabled."
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

**Comment by MarioRgzLpz** [comment]
@sergargar Status extended changes done in commit [588971b](https://github.com/prowler-cloud/prowler/pull/4804/commits/588971ba774811d26b610d62a996ce81d313c6a3)

File: tests/providers/aws/services/ec2/ec2_service_test.py [hunk 3]
```diff
@@ -707,3 +726,19 @@ def test__get_launch_template_versions__(self):
             b64decode(version2.template_data["UserData"]).decode(encoding_format_utf_8)
             == KNOWN_SECRET_USER_DATA
         )
+
+    # Test EC2 Describe VPN Endpoints
+    @mock.patch("botocore.client.BaseClient._make_api_call", new=mock_make_api_call)
+    @mock_aws
```
Comment (MrCloudSec) [inline tests/providers/aws/services/ec2/ec2_service_test.py]: Why this?

**Review by MrCloudSec — DISMISSED** [review]
👏🏼 👏🏼

**Review by MrCloudSec — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**